pkg/agent/multicluster/policy_only_route_controller.go

```diff
@@ -0,0 +1,306 @@
+			DeleteFunc: func(old interface{}) {
+				controller.enqueuePod(old, true)
+			},
+		},
+		resyncPeriod,
+	)
+	return controller
+}
+
+func (c *MCWithPolicyOnlyNodeRouteController) enqueueGateway(obj interface{}, isDelete bool) {
+	gw, isGW := obj.(*mcv1alpha1.Gateway)
+	if !isGW {
+		klog.ErrorS(errors.New("received unexpected object"), "enqueueGateway can't process event", "obj", obj)
+		return
+	}
+
+	if gw.Namespace != c.namespace {
+		return
+	}
+
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	// Gateway webhook will guarantee that at most one Gateway in a cluster.
+	if isDelete {
+		c.activeGWName = ""
+		if c.nodeConfig.Name != gw.Name {
+			klog.InfoS("This Node is not the deleted Gateway, skip it", "node", c.nodeConfig.Name, "gateway", gw.Name)
+			return
+		}
+
+		for podIP := range c.installedPods {
+			c.queue.Add(&endpointChangeInfo{
+				isDelete: true,
+				podIP:    podIP,
+			})
+			delete(c.installedPods, podIP)
+		}
+		return
+	}
+
+	c.activeGWName = gw.Name
+	if c.nodeConfig.Name != gw.Name {
+		klog.InfoS("The Node is not the Gateway, skip it", "node", c.nodeConfig.Name, "gateway", gw.Name)
+		return
+	}
+	pods, _ := c.podLister.List(labels.Everything())
+	for _, p := range pods {
+		pod := p
+		isValidPod := c.validatePod(pod)
+		if !isValidPod {
+			continue
+		}
+		c.queue.Add(&endpointChangeInfo{
+			podIP:     pod.Status.PodIP,
+			podNodeIP: pod.Status.HostIP,
+		})
+		c.installedPods[pod.Status.PodIP] = pod.Status.HostIP
+	}
+}
+
+// validatePod validates the Pod info to determine if it's a valid Pod which should
+// have	Openflow flows installed for cross-cluster traffic.
+func (c *MCWithPolicyOnlyNodeRouteController) validatePod(pod *corev1.Pod) bool {
+	podIP := pod.Status.PodIP
+	podNodeIP := pod.Status.HostIP
+
+	if pod.Spec.NodeName == c.activeGWName {
+		klog.V(2).InfoS("Skip Pod on the Gateway Node")
+		return false
+	}
+
+	if podIP == "" || podNodeIP == "" || pod.Spec.HostNetwork {
+		klog.V(2).InfoS("Pod has no valid IPs or it's HostNetwork, skip it", "name", klog.KObj(pod),
+			"podIP", podIP, "nodeIP", podNodeIP, "hostNetwork", pod.Spec.HostNetwork)
+		return false
+	}
+
+	if c.installedPods[podIP] == podNodeIP {
+		klog.V(2).InfoS("No Pod change impacts installed Openflow rules", "name", klog.KObj(pod))
+		return false
+	}
+	return true
+}
+
+func (c *MCWithPolicyOnlyNodeRouteController) enqueuePod(obj interface{}, isDelete bool) {
+	pod, isPod := obj.(*corev1.Pod)
+	if !isPod {
+		klog.ErrorS(errors.New("received unexpected object"), "enqueuePod can't process event", "obj", obj)
+		return
+	}
+
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	if c.activeGWName == "" || c.nodeConfig.Name != c.activeGWName {
+		klog.InfoS("No active gateway or the Node is not the active Gateway", "node", c.nodeConfig.Name, "gateway", c.activeGWName)
+		return
+	}
+
+	podIP := pod.Status.PodIP
+	podNodeIP := pod.Status.HostIP
+	_, exists := c.installedPods[podIP]
+	if isDelete && exists {
+		c.queue.Add(&endpointChangeInfo{
```

Comment:
No need to set "isDelete", just check if `podIP` does not exist in the `installedPods`, and trigger uninstall process?